Automattic/wp-calypso
# Copy site: add flow validation to check if `sourceSlug` is a valid site

#### Proposed Changes

* Add a new hook to check if a given `sourceSlug` verifies all the requirements to be a copy site origin. If it's not a valid site, then we redirect to `/sites`

#### Testing Instructions



* Disable your proxy to be a normal user.
* Go to http://calypso.localhost:3000/setup/copy-site/domains?sourceSlug=nonexistingweirdsite.wpcomstaging.com
* Observe you are redirected to `/sites`
* Open the actions menu on a business site 
* Click on `Copy Site`
* Observe you see the Domains step, and you can proceed to copy the site as usual.

#### Pre-merge Checklist





- Closes https://github.com/Automattic/dotcom-forge/issues/1503
